### PR TITLE
Include git hash in PDF

### LINKED ISSUE #78: Add timestamp and a link to this GitHub repo
Please add "last modified XXX" somewhere, and a link to this GitHub repo.

http://www.reddit.com/r/ethereum/comments/2x62jn/how_up_to_date_is_the_yellow_paper/

### PR DESCRIPTION
**Pull request by axic**
Fixes #78.

### CODE REVIEW

**Comment by axic** [comment]
If #214 is merged, we could add this to the section:
> This version was generated based on the commit \YellowPaperVersionNumber{} on \YellowPaperBuildTimestamp}

**Comment by axic** [comment]
@chriseth do you mean having the date in the title? If so what date format? (YYYY/MM/DD)

`unknown version` plus a date might not fit into the title well if you consider it is duplicated as the header on every page, but this might fit well into #214.

**Comment by axic** [comment]
This fits as the header on the pages:

`ETHEREUM: A SECURE DECENTRALISED GENERALISED TRANSACTION LEDGER EIP-150 REVISION (d6b72f7 - 2017-01-12)`

**Comment by chriseth** [comment]
I guess it is fine to just put it on the first page.

File: build.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+SHA=`git rev-parse --short --verify HEAD`
+
+echo "\newcommand{\YellowPaperVersionNumber}{$SHA}" > Version.tex
+
+pdflatex -interaction=errorstopmode -halt-on-error Paper.tex
+bibtex Paper
+pdflatex -interaction=errorstopmode -halt-on-error Paper.tex
+pdflatex -interaction=errorstopmode -halt-on-error Paper.tex
+pdflatex -interaction=errorstopmode -halt-on-error Paper.tex
```
Comment (pirapira) [inline build.sh]: I think these command should be concatenated with `&& \`.   Then, if one of the commands fail, Travis can notice.

Comment (axic) [inline build.sh]: Moved the build part to #220 and fixed it there. That PR can be merged I guess right now.

**Comment by axic** [comment]
@chriseth @pirapira added build date (YYY-MM-DD of the `HEAD` commit) and it falls back to unknown without a repo.

File: build.sh [hunk 2]
```diff
@@ -2,6 +2,20 @@
 
 set -e
 
+if [ -d ".git" ]; then
+
+SHA=`git rev-parse --short --verify HEAD`
+DATE=`git show -s --format="%cd" --date="format:%Y-%m-%d" HEAD`
```
Comment (axic) [inline build.sh]: This apparently is not supported on every `git` version. Any ideas how to nicely find out the commit date?

Comment (axic) [inline build.sh]: `git --date=short` does the trick.

**Pull request merged**